dashboard/test/models/script_test.rb

```diff
@@ -931,6 +932,16 @@ def populate_cache_and_disconnect_db
     assert_equal '/resources-pdf-url', summary[:scriptResourcesPdfUrl]
   end
 
+  test 'get_unit_resources_pdf_url returns nil if no resources in script or lessons' do
+    Services::CurriculumPdfs.stubs(:get_unit_resources_url).returns('/resources-pdf-url')
+    unit = create(:script, name: 'single-lesson-script', instruction_type: SharedCourseConstants::INSTRUCTION_TYPE.teacher_led, instructor_audience: SharedCourseConstants::INSTRUCTOR_AUDIENCE.teacher, participant_audience: SharedCourseConstants::PARTICIPANT_AUDIENCE.student)
```

Comment:
Certainly doesn't need to be part of this PR, but it is on EPT's radar to create script factories with the new instruction types? Kind of like how we have ["shortcut" factories](https://github.com/code-dot-org/code-dot-org/blob/0ca01460d647b30a70f583f5930f7e790b5f3407/dashboard/test/factories/factories.rb#L779) to make scripts of different CSX types? Could be one way to keep tests a tad tidier and use factories as sort of documentation for the different script types.

Reply:
great call @Erin007 ! +1 to setting up more factory shortcuts. we have not been great about this in our area in general, for example I just added a test with a bunch of copy-pasted lines just to have a script containing a level: https://github.com/code-dot-org/code-dot-org/blob/f75b1b6db38882874b181c051e3062a0418583ec/dashboard/test/controllers/scripts_controller_test.rb#L1654-L1660